zaptest/observer/observer.go

```diff
@@ -80,6 +80,39 @@ func (o *ObservedLogs) AllUntimed() []LoggedEntry {
 	return ret
 }
 
+// FilterMessage filters entries to those that have the specified message.
+func (o *ObservedLogs) FilterMessage(msg string) *ObservedLogs {
+	return o.filter(func(e LoggedEntry) bool {
+		return e.Message == msg
+	})
+}
+
+// FilterField filters entries to those that have the specified field.
+func (o *ObservedLogs) FilterField(field zapcore.Field) *ObservedLogs {
+	return o.filter(func(e LoggedEntry) bool {
+		for _, ctxField := range e.Context {
+			if ctxField.Key != field.Key {
+				continue
+			}
+			return ctxField == field
```

Comment:
`continue` vs `return` -- if we find a field with the right key, then the match is whether the field matches exactly. If the field mismatches on keys, we want to check other fields (since the filter only needs one field to match exactly).